### ISSUE TITLE
OCIO/OIIO CMake conflict when using OpenEXR >= 2.5 for Half dependency

### ISSUE THREAD

**Issue by michdolan** [AcademySoftwareFoundation/OpenColorIO#1261]
I've only tested this on Windows so far, but while updating my Windows builds of OCIO v2.0.0-rc1 and OIIO 2.2.10.1, both linking against OpenEXR 2.5.4, this fails the CMake configuration.

The cause is that we are relying on OIIO's CMake config to find OIIO, which has a condition which relies on and calls OpenEXR's CMake config when using OpenEXR >=2.5. Since OCIO already has a target for Half, but not for the other OpenEXR components it creates a conflict and fails to setup the OIIO dependency.

I'm actively working on removing the OIIO dependency from OCIO, which will solve this issue, but I wanted to mention it. We could resolve it on our end by either building all of OpenEXR (instead of just Half), or by reverting to using our old OIIO find module instead of the config.

**Comment by michdolan** [comment]
```
CMake Error at .../openexr-2.5.4/lib/cmake/IlmBase/IlmBaseTargets.cmake:37 (message):
  Some (but not all) targets in this export set were already defined.

  Targets Defined: IlmBase::Half

  Targets not yet defined:
  IlmBase::IlmBaseConfig;IlmBase::Iex;IlmBase::IexMath;IlmBase::Imath;IlmBase::IlmThread


Call Stack (most recent call first):
  .../openexr-2.5.4/lib/cmake/IlmBase/IlmBaseConfig.cmake:27 (include)
  C:/Program Files/CMake/share/cmake-3.16/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  .../openexr-2.5.4/lib/cmake/OpenEXR/OpenEXRConfig.cmake:30 (find_dependency)
  C:/Program Files/CMake/share/cmake-3.16/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  .../oiio-2.2.10.1/lib/cmake/OpenImageIO/OpenImageIOConfig.cmake:31 (find_dependency)
  src/CMakeLists.txt:11 (find_package)


-- Configuring incomplete, errors occurred!
```

**Comment by remia** [comment]
I think I stumbled on the same issue, on macOS using `OIIO 2.2.10.1` and `OpenEXR 2.5.4`, here it complains about the Threads target. This is repeated for `ociolutimage` `ocioconvert` `ociodisplay` `ocioperf` `oiiohelpers`.

```
CMake Error at src/apps/ociolutimage/CMakeLists.txt:8 (add_executable):
  Target "ociolutimage" links to target "Threads::Threads" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```

Not so related but when I tried to configure with `-DOCIO_BUILD_APPS=OFF`, I got a compilation error while building the `test_gpu_exec` target:

```
.../OpenColorIO/tests/gpu/GPUUnitTest.cpp:18:10: fatal error: 'oglapp.h' file not found
```

My current understanding is that the condition for the oglapphelpers target include in the build process may not be correct, https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/master/src/libutils/CMakeLists.txt#L4 at this point in the normal process, it seems that `test_gpu_exec` is not defined no matter what. This is because the `tests` directory is added after `src` (that itself includes `libutils`) here https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/master/CMakeLists.txt#L255. This is just a quick analysis, don't take my word for granted though.
